libmuscle/cpp/src/libmuscle/profiler.cpp

```diff
@@ -0,0 +1,33 @@
+#include <libmuscle/profiler.hpp>
+
+#include <libmuscle/profiling.hpp>
+
+
+namespace libmuscle { namespace impl {
+
+Profiler::Profiler(MMPClient & manager)
+    : manager_(manager)
+    , events_()
+{}
+
+void Profiler::shutdown() {
+    flush_();
+}
+
+void Profiler::record_event(ProfileEvent && event) {
+    if (!event.stop_time.is_set())
+        event.stop_time = ProfileTimestamp();
+    events_.push_back(std::move(event));
+    if (events_.size() >= 100)
```

Comment:
Yeah, temporary solution :-). It's probably best to do it based on a fixed time interval, and I'd like a scale factor for that as well eventually that the manager can set, so that if there are very many instances it can tell everyone to slow down a bit more. Although at some point the per-call overhead is negligible and we're just swamping the manager, so maybe that's not needed. Anyway, we'll fix that when they finish that European exascale machine :-). I've added an item to https://github.com/multiscale/muscle3/issues/78 for this, it can be fixed after we merge this but before the next release.